lib/fog/google/models/pubsub/subscriptions.rb

```diff
@@ -0,0 +1,33 @@
+require "fog/core/collection"
+require "fog/google/models/pubsub/subscription"
+
+module Fog
+  module Google
+    class Pubsub
+      class Subscriptions < Fog::Collection
+        model Fog::Google::Pubsub::Subscription
+
+        # Lists all subscriptions that exist on the project.
+        #
+        # @return [Array<Fog::Google::Pubsub::Subscription>] list of
+        #   subscriptions
+        def all
+          data = service.list_subscriptions.body["subscriptions"] || []
+          load(data)
+        end
+
+        # Retrieves a subscription by name
+        #
+        # @param subscription_name [String] name of subscription to retrieve
+        # @return [Fog::Google::Pubsub::Topic] topic found, or nil if not found
+        def get(subscription_name)
+          if subscription = service.get_subscription(subscription_name).body
```

Comment:
Conditional assignment. This is confusing and will always get called if `service.get_subscription(subscription_name).body` doesn't return an exception.